observation_portal/requestgroups/duration_utils.py

```diff
@@ -258,13 +258,15 @@ def get_request_duration_by_instrument_type(request_dict):
     durations_by_instrument_type = get_complete_configurations_duration_by_instrument_type(
         configurations, start_time)
 
-    # Add in the front_padding proportionally by instrument_type here
-    # TODO: We should move front_padding to the telescope level rather than instrument_type so we don't need to
-    # assign it's value proportionally to observation time used per instrument_type
     total_duration = 0.0
-    for duration in durations_by_instrument_type.values():
-        total_duration += duration
     for instrument_type in durations_by_instrument_type.keys():
+        # First multiply the configuration durations by the configuration_repeats count
+        durations_by_instrument_type[instrument_type] *= request_dict.get('configuration_repeats', 1)
+        total_duration += durations_by_instrument_type[instrument_type]
+    for instrument_type in durations_by_instrument_type.keys():
```

Comment:
Maybe I'm misunderstanding, but do we need two for loops here? It seems to me we could do everything for each instrument type in a single loop? Or is there some subtlety I'm missing?

Reply:
Yeah, the first loop is to populate the total_duration including all instrument types. We need the final total_duration to use in the second loop to calculate the proportional amount of time per instrument_type, which we use to give the proper proportion of front_padding time to each instrument times duration. Like the TODO comment says, if we move the front_padding from an instrument_type to the telescope level in configdb, then we wouldn't need to do these loops.